jetty-servlets/src/main/java/org/eclipse/jetty/servlets/HeaderFilter.java

```diff
@@ -116,6 +114,8 @@ public void doFilter(ServletRequest request, ServletResponse response, FilterCha
                 }
             }
         }
+
+        chain.doFilter(request,response);
```

Comment:
If shouldfilter returns false, then doFilter is never called because of the return at line 88 above.